hedera-node/hedera-mono-service/src/main/java/com/hedera/services/grpc/marshalling/ImpliedTransfersMarshal.java

```diff
@@ -89,7 +89,9 @@ public ImpliedTransfers unmarshalFromGrpc(CryptoTransferTransactionBody op, Acco
             final var aliasResolver = aliasResolverFactory.get();
             op = aliasResolver.resolve(op, aliasManager);
             numAutoCreations = aliasResolver.perceivedAutoCreations();
-            if (numAutoCreations > 0 && !props.isAutoCreationEnabled()) {
+            if (numAutoCreations > 0
+                    && !props.isAutoCreationEnabled()
+                    && !props.isLazyCreationEnabled()) {
```

Comment:
I have a small concern here:

Won't this return `NOT_SUPPORTED` in the case where:
1. We have 1 or more transfers to non-existing **key aliases**, which will trigger **auto-creation** (not lazy creation)
2. Auto creation flag **enabled**, lazy creation flag **disabled**?

In other words, we won't be able to execute the auto-creation flow if the lazy creation flag is disabled given this, correct?

If so, maybe we need to have a separate field in the `aliasResolver` to track the lazy creations, something like `perceivedLazyCreations`?

Reply:
Won't this check solve this issue? 

`((numAutoCreations > 0 && !props.isAutoCreationEnabled()) || !props.isLazyCreationEnabled())`

Reply:
This would make for a good test case scenario for the test plan.

Reply:
I think it will return `OK` in this case, [I have added a test for it](https://github.com/hashgraph/hedera-services/blob/03953-crypto-transfer-with-evm-address-alias/hedera-node/hedera-mono-service/src/test/java/com/hedera/services/grpc/marshalling/ImpliedTransfersMarshalTest.java#L207).

Reply:
> Won't this check solve this issue?
> 
> `((numAutoCreations > 0 && !props.isAutoCreationEnabled()) || !props.isLazyCreationEnabled())`

For the case we are considering this check will evaluate to `((true && false) || true)` = `true` and it will return `NOT_SUPPORTED` which is not what we want.

Reply:
Oh, correct, my mistake here, the problem is different than what I first assumed 😄 

Given auto creation = true, lazy creation = false
1. In `AliasResolver` we[ count lazy creations toward the `perceivedCreations` counter,](props.isLazyCreationEnabled()) shared with the auto creations, so we can't know only based from the `numAutoCreations` field whether we have lazy creates, auto creates, or both; 
2. In this condition statement, even if there are lazy creations in the op body, but the lazy creation flag is not enabled, we won't return invalid, since the auto-create flag is true
5. We will continue down the logic, until we reach AutoCreationLogic, where [here we will enter the else, since the lazy create flag is false](https://github.com/hashgraph/hedera-services/pull/4004/files#diff-ef751d86c731f6719ae00b599f79822e36498693043b9d29ea0b6e6901fa8cdeR204), and an [exception will be thrown when we attempt to convert the `evmAddress` to e protobuf `Key`](https://github.com/hashgraph/hedera-services/pull/4004/files#diff-ef751d86c731f6719ae00b599f79822e36498693043b9d29ea0b6e6901fa8cdeR216)

There will probably be problems also in the scenario where auto creation = false, lazy creation = true, where we should not allow any auto creations (normal or lazy), but we will still pass this condition and continue down the logic.

Does that make sense or am I missing something again? 😄

Reply:
Added implementation to track the number of lazy creations separately from the one for auto creations.